src/Compilers/Core/Portable/TreeDumper.cs

```diff
@@ -106,6 +102,31 @@ private void DoDumpCompact(TreeDumperNode node, string indent)
                 // the child node's children:
                 DoDumpCompact(child, indent + (i == children.Count - 1 ? "  " : "\u2502 "));
             }
+
+            static IEnumerable<TreeDumperNode> filter(IEnumerable<TreeDumperNode> nodes)
```

Comment:
>filter [](http://example.com/codeflow?start=47&length=6)

Consider changing this method to `bool filter(TreeDumperNode node)` and using `node.Children.Where(filter)` above.